echopop/utils/load.py

```diff
@@ -575,3 +578,58 @@ def prepare_input_data(input_dict: dict, configuration_dict: dict):
 
     # Return updated dictionaries
     return input_dict, configuration_dict
+
+
+def validate_config_structure(yaml_data, config_spec):
+    """
+    Validate configuration YAML dictionary structure and entry types
+    """
+
+    # Helper function for validating dictionary key names/structure
+    def validate_dict(data, spec, path=""):
+        for key, value in spec.items():
+            current_path = f"{path}.{key}" if path else key
+            if key == "ANY":
+                # If the spec key is "ANY", check all keys in the data against the "ANY" spec
+                if not isinstance(data, dict):
+                    raise ValueError(
+                        f"Expected a dictionary at {current_path}, got {type(data).__name__}"
+                    )
+                for sub_key in data:
+                    validate_value(data[sub_key], spec[key], f"{current_path}.{sub_key}")
+            else:
+                if key not in data:
+                    raise KeyError(f"Missing key: {current_path}")
+                validate_value(data[key], value, current_path)
+
+    # Helper function for parsing values entered at different points throughout the dictionary
+    def validate_value(data, spec, path):
+        if isinstance(spec, dict):
+            validate_dict(data, spec, path)
+        elif isinstance(spec, type):
+            if not is_valid_type(data, spec):
+                raise TypeError(
+                    f"Expected type {spec.__name__} at {path}, got {type(data).__name__}"
+                )
+        elif isinstance(spec, list):
+            if not isinstance(data, list):
+                raise TypeError(f"Expected a list at {path}, got {type(data).__name__}")
+            if len(spec) != 1:
+                raise ValueError(
+                    f"Spec list at {path} should contain exactly one type element, got {spec}"
+                )
+            for index, item in enumerate(data):
+                validate_value(item, spec[0], f"{path}[{index}]")
+        else:
+            raise ValueError(f"Unknown spec type at {path}: {spec}")
+
+    # Helper function for assessing the data type of each nested value
+    def is_valid_type(data, spec_type):
+        if spec_type is float:
+            return isinstance(data, (int, float))
```

Comment:
I think a better handling here is probably to coerce the conversion from `int` to `float` and then spit out a warning to let user know about it. This check here can lead to bugs that are hard to check, because you are not checking the type as it is written.